src/layers/PolygonSeries.js

```diff
@@ -111,10 +111,14 @@ export class PolygonSeries extends PaletteMixin(CoverageMixin(L.Layer)) {
     if (t.coordPref == undefined) {
       t.idx = t.coord = undefined
     } else {
-      let vals = this.domain.axes.get('t').values.map(v => v.getTime())
-      t.idx = indexOfNearest(vals, t.coordPref.getTime())
+      let vals = this.domain.axes.get('t').values.map(v => asTime(v))
+      t.idx = indexOfNearest(vals, asTime(t.coordPref))
       t.coord = vals[t.idx]
     }
+
+    this.coverage.loadRange(this.parameter.key).then(range => {

```

Comment:
Is this temporary? Does look like it belongs here.

Reply:
No, that's required.  Without it, `this.range` isn't available for getting the value / palette extent.  I copied it from `CoverageMixin.js` where it gets called if `this._loadCoverageSubset` doesn't exist.  I figured that since it seems to be necessary, and was not getting called if `this._loadCoverageSubset` is present, then that should be the method that takes care of loading the range.  Happy to move it somewhere else if you have a suggestion (or handle the problem some other way).

Reply:
I think the best solution would be within `PolygonSeries` (and similar degenerate cases which would need a `loadRange()`) to rename `_loadCoverageSubset` to something else like `_updateSubsetInfo` such that `CoverageMixin` doesn't skip `loadRange()`. It's not necessarily wrong to keep it like you suggested and add `loadRange` but then you need to do more work and also be careful to handle all the async cases, and the loading events. Also, calling `loadRange()` unnecessarily multiple times (at each `set time`) is wasteful without caching, which could be solved but I think it's not worth the effort.